UnityProject/Assets/Scripts/Systems/Chemistry/Reaction.cs

```diff
@@ -27,11 +27,11 @@ public class Reaction : ScriptableObject
 		public float? tempMin;
 		public float? tempMax;
 
-		public virtual bool Apply(MonoBehaviour sender, ReagentMix reagentMix)
+		public virtual bool Apply(object sender, ReagentMix reagentMix)
 		{
 			if (IsReactionValid(reagentMix) == false) return false;
 
-			ApplyReaction(sender, reagentMix);
+			ApplyReaction(sender as MonoBehaviour, reagentMix);
```

Comment:
why do you need to get it as object then?

Reply:
because systems are just classes

Reply:
but you are casting this to MonoBehaviour. Uhm?

Reply:
in the overrides it is doing as